### PR TITLE
Logout improvements

### PR DESCRIPTION
**Pull request by airbornemint**
Implements the following funcionality:

 * Added `logout_url` to `oauth` conf section
 * Request to `/logout` will be redirected to `oauth.logout_url` conf if the latter is present
 * Token is included in redirect to `oauth.logout_url ` in the `id_token_hint ` URL param
 * If `oauth.logout_url` is present and the `/logout` request also contains a `url` param, then the redirect to `oauth.logout_url` will include the url in the `post_logout_redirect_uri ` param in it as well. 

In other words:

 * `url` param in `/logout` is to be used for *app* URL that the user should be redirected to after logout
 * `oauth.logout_url` conf is to be used for logging out of the OAuth provider

See #40 #231

### CODE REVIEW

**Comment by bnfinet** [comment]
@airbornemint thank you for the contribution

Can we please continue the conversation in #40

File: handlers/common/common.go [hunk 1]
```diff
@@ -4,23 +4,25 @@ import (
 	"context"
```
Comment (bnfinet) [inline handlers/common/common.go:4]: common.go is "common to providers" and was created when all of the individual provider packages were created (adfs.go, oidc.go).  all of these `providers` are going to move soon, probably to `/pkg/providers/*`

`FindJWT` and `ClaimsFromJWT` do need a better home but I don't think this is it.  Perhaps `jwtmanager` would be better.

File: handlers/logout.go [hunk 2]
```diff
@@ -27,17 +38,48 @@ func LogoutHandler(w http.ResponseWriter, r *http.Request) {
...
+		if (!redirectValid) {
+			error400(w, r, fmt.Errorf("%w: %s", errUnauthRedirURL, redirectURL))
+			return
+		}
+	}
+
+	// If provider logout URL is configured, redirect to it (and pass redirectURL along)
+	// If provider logout URL is not configured, redirect directly to redirectURL
+	if providerLogoutURL != "" {
+		req, err := http.NewRequest("GET", providerLogoutURL, nil)
```
Comment (bnfinet) [inline handlers/logout.go:63]: Instead of using `http.NewRequest` please use `url.Parse` similar to this implementation...
https://github.com/coreos/go-oidc/pull/190/files

File: pkg/cfg/cfg.go [hunk 3]
```diff
@@ -73,6 +73,7 @@ type oauthConfig struct {
 	ClientSecret    string   `mapstructure:"client_secret"`
 	AuthURL         string   `mapstructure:"auth_url"`
 	TokenURL        string   `mapstructure:"token_url"`
+	LogoutURL       string   `mapstructure:"logout_url"`
```
Comment (bnfinet) [inline pkg/cfg/cfg.go:76]: I'm not sure but this might better be `end_session_endpoint`
https://developer.okta.com/docs/reference/api/oidc/#logout end_session_endpoint
https://openid.net/specs/openid-connect-session-1_0-17.html

WDYT?

Comment (airbornemint) [reply pkg/cfg/cfg.go:76]: Based on the precedent from `configureOAuthClient`, he existing nomenclature is that individual API endpoints are identified by things named `*URL`, and that a collection of related API endpoints is called `Endpoint`. So I would keep `_url` here for consistency, but I think that `end_session_url` would be fine. WDYT?

File: handlers/common/common.go (hunk 1 above)
Comment (airbornemint) [reply handlers/common/common.go]: Done

File: handlers/logout.go (hunk 2 above)
Comment (airbornemint) [reply handlers/logout.go]: Done

**Comment by bnfinet** [comment]
sorry for the moving target @airbornemint I'll take care of those adjustments when I look at this next, hopefully tomorrow or early next week.

Thanks again for the code!

**Comment by bnfinet** [comment]
After more research I think we do need to go with `end_session_endpoint` as per the spec.  For #41 OIDC Discovery it'll need to comply.

* [Okta](https://developer.okta.com/docs/reference/api/oidc/#well-known-oauth-authorization-server)
* Google
* [ADFS](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/development/ad-fs-logout-openid-connect)
* not sure about [Auth0](https://community.auth0.com/t/missing-end-session-endpoint-in-well-known-openid-configuration/31260) but I suspect they will soon :)

I'm going to make the change as I merge/publish
